gammapy/spectrum/models.py

```diff
@@ -1464,3 +1465,61 @@ def evaluate(self, energy, **kwargs):
         flux = self.spectral_model.evaluate(energy=energy, **kwargs)
         absorption = self.absorption.evaluate(energy=energy, parameter=parameter)
         return flux * absorption
+
+
+class NaimaModel(SpectralModel):
+    r""""""
+    __slots__ = ["radiative_model", "distance"]
+
+    def __init__(self, radiative_model, distance=1.0 * u.kpc):
+        self.radiative_model = radiative_model
+        self.distance = Parameter("distance", distance)
+
+        parameters = []
+        parameters_dict = self.radiative_model.particle_distribution.__dict__
+        for (name, quantity) in parameters_dict.items():
+            if name[0] == "_" or name == "unit":
+                continue
+            parameters.append(Parameter(name, quantity))
+
+        super().__init__(parameters)
+
+    def __call__(self, energy, seed=None, distance=None):
+        """Call evaluate method"""
+        kwargs = dict()
+        for par in self.parameters.parameters:
+            quantity = par.quantity
+            if quantity.unit.physical_type == "energy":
+                quantity = quantity.to(energy.unit)
+            kwargs[par.name] = quantity
+
+        if distance == None:
+            distance = self.distance.quantity
+        eval = self.evaluate(
+            energy.flatten(), self.radiative_model, seed, distance, kwargs
+        )
+
+        return eval.reshape(energy.shape)
+
+    def freeze(self, name, freeze=True):
+        """"""
+        try:
+            par_idx = self.parameters._get_idx(name)
+        except:
+            raise AttributeError("Parameter {} not found".format(name))
+
+        self.parameters.parameters[par_idx].frozen = freeze
+
+    @staticmethod
+    def evaluate(energy, radiative_model, seed, distance, kwargs):
+        """Evaluate the model (static function)."""
+        parameters_dict = radiative_model.particle_distribution.__dict__
+        for key, value in kwargs.items():
+            parameters_dict[key] = kwargs.get(key, value)
+
+        if seed == None:
+            dnde = radiative_model.flux(energy, distance=distance)
+        else:
+            dnde = radiative_model.flux(energy, seed=seed, distance=distance)
+
+        return dnde.to("cm-2 s-1 TeV-1")
```

Comment:
I think that it would be nice if the energy part of the flux was returned in the same units as the input energy array. So for example, if I compute the model using an array like `[1.00000000e-02, 3.72759372e-02, 1.38949549e-01] MeV` I get a flux in units of cm-2 s-1 MeV-1. Do you think this might make sense? If not, I will just drop any specification and leave it to the default units (which are cm-2 s-1 eV-1)

Reply:
Yes, I agree this would make sense from a user-perspective. You can achieve this by creating a composite unit like so `unit = 1 / (u.s * u.sr * energy.unit)`.